src/web/core2numpy.rkt

```diff
@@ -0,0 +1,111 @@
+  (match (cons op args)
+   [(list 'deg2rad a) (format "numpy.deg2rad(~a)" a)]
+   [(list 'rad2deg a) (format "numpy.rad2deg(~a)" a)]
+   [(list 'logaddexp a b) (format "numpy.logaddexp(~a, ~a)" a b)]
+   [(list 'logaddexp2 a b) (format "numpy.logaddexp2(~a, ~a)" a b)]
+   [(list 'square a) (format "numpy.square(~a)" a)]
+   [(list 'asin a) (format "numpy.arcsin(~a)" a)]
+   [(list 'acos a) (format "numpy.arccos(~a)" a )]
+   [(list 'atan a) (format "numpy.arctan(~a)" a )]
+   [(list 'asinh a) (format "numpy.arcsinh(~a)" a )]
+   [(list 'acosh a) (format "numpy.arccosh(~a)" a )]
+   [(list 'atanh a) (format "numpy.arctanh(~a)" a )]
+   [(list 'atan2 a b) (format "numpy.arctan2(~a, ~a)" a b)]
+   [(list '+ a b) (format "numpy.add(~a, ~a)" a b)]
+   [(list 'recip a) (format "numpy.reciprocal(~a)" a )]
+   [(list 'neg a) (format "numpy.negative(~a)" a )]
+   [(list '* a b) (format "numpy.multiply(~a, ~a)" a b)]
+   [(list '/ a b) (format "numpy.divide(~a, ~a)" a b)]
+   [(list 'pow a b) (format "numpy.float_power(~a, ~a)" a b)]
+   [(list '- a b) (format "numpy.subtract(~a, ~a)" a b)]
+   [(list '- a) (format "numpy.negative(~a)" a)]
+   [(list 'rint a) (format "numpy.rint(~a)" a )] 
+   [(list '> a b) (format "numpy.greater(~a, ~a)" a b)]
+   [(list '>= a b) (format "numpy.greater_equal(~a, ~a)" a b)]
+   [(list '< a b) (format "numpy.less(~a, ~a)" a b)]
+   [(list '<= a b) (format "numpy.less_equal(~a, ~a)" a b)]
+   [(list '== a b) (format "numpy.equal(~a, ~a)" a b)]
+   [(list '!= a b) (format "numpy.not_equal(~a, ~a)" a b)]
+   [_ (format "numpy.~a(~a)" op (string-join args ", "))]))
+
+(define (constant->numpy x ctx)
+  (define prec (ctx-lookup-prop ctx ':precision))
+  (match x
+   ['TRUE "True"]
+   ['FALSE "False"]
+   ['INFINITY (format "numpy.ones(length, dtype=~a)*numpy.inf" (type->numpy prec))]
+   ['NAN (format "numpy.ones(length, dtype=~a)*numpy.nan" (type->numpy prec))]
+   ['PI (format "numpy.ones(length, dtype=~a)*numpy.pi" (type->numpy prec))]
+   ['E (format "numpy.ones(length, dtype=~a)*numpy.e" (type->numpy prec))]
+   [(? hex?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) (real->double-flonum (hex->racket x)))]
+   [(? number?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) (real->double-flonum x))]
+   [(? symbol?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) x)]))
+
+(define declaration->numpy
+  (case-lambda
+   [(var ctx)
+    (define prec (ctx-lookup-prop ctx ':precision))
+    (format "~a = ~a" var (match prec ['binary64 "numpy.zeros(length)"] ['binary32 "numpy.zeros(length)"] ['boolean "True"]))]
+   [(var val ctx)
+    (format "~a = ~a" var val)]))
+
+(define (assignment->numpy var val ctx)
+  (format "~a = ~a" var val))
+
+(define (program->numpy name args arg-ctxs body ret ctx used-vars)
+  (format "def ~a(~a):\n\tlength = ~a.shape\n~a\treturn ~a\n" name
+          (string-join args ", ") (if (> (length args) 0) (first args) "numpy.ones(5)") body ret))
```

Comment:
Uhh what's with the default output here??

Reply:
I guess we're just saying it doesn't matter what we return? In Python you can skip the `return` line and just not return anything.

Reply:
Are you talking about the `numpy.ones(5)`? I wasn't sure what to set the length to if we don't have any parameters for this method. Should I do something different instead?

Reply:
It's probably fine though maybe instead of 5 we should use the actual length we use by default?

Reply:
What is the actual length we use by default?

Reply:
Didn't you write the runner 🤣

Reply:
Oh yeah, nevermind thanks :thumbsup: